nevergrad/functions/rl/agents.py

```diff
@@ -129,10 +129,11 @@ def __init__(
     ) -> None:
         assert isinstance(env_runner.env, gym.Env)
         self.agent = agent.copy()
-        self.runner = env_runner
+        self.runner = env_runner.copy()
         self.reward_postprocessing = reward_postprocessing
-        super().__init__(self.compute, **self.agent.instrumentation.kwargs)
-        self._descriptors.update(num_repetitions=self.runner.num_repetitions, instrumentation="")
+        super().__init__(self.compute, self.agent.instrumentation.copy().with_name(""))
+        self.register_initialization(agent=agent, env_runner=env_runner, reward_postprocessing=reward_postprocessing)
+        self._descriptors.update(num_repetitions=self.runner.num_repetitions, archi=self.agent.module.__class__.__name__)
```

Comment:
I've added the `archi` descriptor here. It takes the module class name which is `Perceptron` or `DenseNet`. This will replace `mono` and `multi`  which were manually added in the experiments (which does not work anymore -> descriptors should preferably be automatic to avoid errors)